tests/unit/core/api/test_index.js

```diff
@@ -235,7 +239,40 @@ describe(__filename, () => {
 
     it('sets the app, lang, and type query', () => {
       mockWindow.expects('fetch')
-        .withArgs(`${apiHost}/api/v3/addons/featured/?app=android&type=persona&lang=en-US`)
+        .withArgs(`${apiHost}/api/v3/addons/featured/?app=firefox&type=persona&lang=en-US`)
+        .once()
+        .returns(mockResponse());
+
+      const { state } = dispatchClientMetadata({
+        clientApp: CLIENT_APP_FIREFOX,
+        lang: 'en-US',
+      });
+
+      return api.featured({
+        api: state.api,
+        filters: { addonType: ADDON_TYPE_THEME },
+      })
+        .then((response) => {
+          expect(response).toEqual({
+            entities: {
+              addons: {
+                foo: { slug: 'foo' },
+                food: { slug: 'food' },
+                football: { slug: 'football' },
+              },
+            },
+            result: {
+              results: ['foo', 'food', 'football'],
+            },
+          });
+          return mockWindow.verify();
+        });
+    });
+
+    it('sets app to `firefox` when addon type is theme for Android', () => {
```

Comment:
Github makes it hard to comment on this new test you added but I think it does too much. All it should do is add the `.withArgs('.../?app=firefox...')` assertion. It doesn't need to check the entities in the response because that's already covered.

To answer the question about mock injection: I think your approach here is fine. This will serve as a smoke test to make sure that the function is calling the helper but it won't check all edge cases.

Reply:
Oh, that's what you meant by smoke test! I thought it had something to do with an injected mock or so... Sorry, I completely misunderstood you. Everything you say here makes sense, going to change that!